service/collector/helper.go

```diff
@@ -183,3 +181,36 @@ func (h *helper) AWSAccountID(awsClients clientaws.Clients) (string, error) {
 
 	return accountID, nil
 }
+
+// ListReconciledClusters provides names of clusters reconciled by this
+// particular operator version. The names are returned in format of a map,
+// instead of a list, for cheaper lookup.
+func (h *helper) ListReconciledClusters() (reconciled map[string]bool) {
+	ctx := context.Background()
+
+	clusterList := &infrastructurev1alpha2.AWSClusterList{}
+	h.clients.CtrlClient().List(
+		ctx,
+		clusterList,
+		runtimeclient.MatchingLabels{
+			collidingOperatorLabel: project.Version(),
+		},
+	)
+	for _, item := range clusterList.Items {
+		reconciled[item.GetName()] = true
+	}
+
+	configList := &providerv1alpha1.AWSConfigList{}
```

Comment:
This here is `aws-operator@master` which does not deal with the legacy types anymore. We do not need this here.

Reply:
That's great, I'm happy dropping this bit.